src/impl/windows/external_commit_helper.cpp

```diff
@@ -17,69 +17,65 @@
 ////////////////////////////////////////////////////////////////////////////
 
 #include "impl/external_commit_helper.hpp"
-
 #include "impl/realm_coordinator.hpp"
 
 #include <algorithm>
-#include <codecvt>
 
 using namespace realm;
 using namespace realm::_impl;
 
-static HANDLE CreateNotificationEvent(std::string realm_path)
-{
-    // replace backslashes because they're significant in object namespace names
-    std::replace(realm_path.begin(), realm_path.end(), '\\', '/');
-
-    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
-    std::wstring path(L"Local\\" + converter.from_bytes(realm_path));
-
-    HANDLE event = CreateEventEx(nullptr, path.c_str(), CREATE_EVENT_MANUAL_RESET, SYNCHRONIZE | EVENT_MODIFY_STATE);
-    if (event == nullptr) {
-        throw std::system_error(GetLastError(), std::system_category());
-    }
-
-    return event;
-}
-
 ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
 : m_parent(parent)
-, m_event(CreateNotificationEvent(parent.get_path()))
-, m_close_mutex(CreateMutexEx(nullptr, nullptr, CREATE_MUTEX_INITIAL_OWNER, SYNCHRONIZE | MUTEX_MODIFY_STATE))
 {
+    std::string path = parent.get_path();
+    std::replace(path.begin(), path.end(), '\\', '/');
+    std::wstring shared_memory_name = L"Local\\Realm_ObjectStore_ExternalCommitHelper_SharedCondVar_" + std::wstring(path.begin(), path.end());
+#if REALM_WINDOWS
+    m_shared_memory = CreateFileMappingW(INVALID_HANDLE_VALUE, nullptr, PAGE_READWRITE, 0, sizeof(*m_condvar_shared), shared_memory_name.c_str());
+    auto error = GetLastError();
+    m_condvar_shared = reinterpret_cast<InterprocessCondVar::SharedPart*>(MapViewOfFile(m_shared_memory, FILE_MAP_ALL_ACCESS, 0, 0, sizeof(*m_condvar_shared)));
```

Comment:
Won't this crash if `CreateFileMappingW` failed? I would expect the error checking to need to appear before code using the result.